src/librustdoc/clean/simplify.rs

```diff
@@ -91,6 +96,14 @@ pub(crate) fn merge_bounds(
             return false;
         }
         let last = trait_ref.trait_.segments.last_mut().expect("segments were empty");
+
+        trait_ref.generic_params.append(&mut bound_params);
+        // Since the parameters (probably) originate from `tcx.collect_*_late_bound_regions` which
+        // returns a hash set, sort them alphabetically to guarantee a stable and deterministic
+        // output (and to fully deduplicate them).
+        trait_ref.generic_params.sort_unstable_by(|p, q| p.name.as_str().cmp(q.name.as_str()));
+        trait_ref.generic_params.dedup_by(|p, q| p.name.as_str() == q.name.as_str());
```

Comment:
```suggestion
        trait_ref.generic_params.dedup_by_key(|p| p.name);
```

Note to self: Regardless of whether rust-timer is going to report a performance regression or not, I am going to change this line for increased performance. Contrary to the `sort_unstable_by` line which requires comparison between strings, we definitely don't need to access the underlying value of the interned strings here (the whole point of interned strings, duh! 😅).

Reply:
If they're symbol, you need to access the underlying string as it is otherwise not the sort you might expect. Unless you don't care it's alpha sorted?